pkg/cli/alpha/scaffold.go

```diff
@@ -0,0 +1,257 @@
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store"
+	"sigs.k8s.io/kubebuilder/v3/pkg/config/store/yaml"
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+)
+
+// NewScaffoldCommand return a new scaffold command
+func NewScaffoldCommand() *cobra.Command {
+	var projectConfig, outputPath string
+	scaffoldCmd := &cobra.Command{
+		Use:   "scaffold",
+		Short: "Re-scaffold an existing kuberbuilder project",
+		Long: `This command is a helper for you upgrade your project to the latest versions scaffold.
+		
+		It will:
+			- Create a new directory name after the project in the current working directory or at the output path
+			- Re-generate the whole project based on the Project file data
+
+		You will still need to move your code and other customizations over.`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			return validate(projectConfig, outputPath)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			cwd, err := os.Getwd()
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			projectPath := getProjectPath(cwd, projectConfig)
+			store := yaml.New(machinery.Filesystem{FS: afero.NewOsFs()})
+			if err = store.LoadFrom(projectPath); err != nil {
+				log.Fatal(err)
+			}
+
+			outputDirectory := getDefaultOutputPath(cwd, store.Config().GetProjectName(), outputPath)
+			// create output directory
+			if err = os.MkdirAll(outputDirectory, os.ModePerm); err != nil {
+				log.Fatal(err)
+			}
+			// use the new directory to set up the new project
+			if err = os.Chdir(outputDirectory); err != nil {
+				log.Fatal(err)
+			}
+			// change back to the cwd after completion
+			defer func() {
+				_ = os.Chdir(cwd)
+			}()
+
+			// init project with plugins
+			if err = kubebuilderInit(store); err != nil {
+				log.Fatal(err)
+			}
+			// call edit subcommands
+			if err = kubebuilderEdit(store); err != nil {
+				log.Fatal(err)
+			}
+			// recreate api/webhooks
+			if err = kubebuilderCreate(store); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	scaffoldCmd.Flags().StringVar(&projectConfig, "project-config", "",
+		"path to a kubebuilder project file if not in the current working directory")
+	scaffoldCmd.Flags().StringVar(&outputPath, "output", "",
+		"path to output the scaffolding. defaults to the current working directory")
+
+	return scaffoldCmd
+}
+
+func kubebuilderCreate(store store.Store) error {
+	resources, err := store.Config().GetResources()
+	if err != nil {
+		return err
+	}
+
+	var webhooks []resource.Resource
+	for i, r := range resources {
+		if r.Webhooks != nil {
+			// you cannot create a webhook without an API, create them after
+			webhooks = append(webhooks, resources[i])
+		} else {
+			if err = createAPI(r); err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, w := range webhooks {
+		if err = createWebhook(w); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createAPI(resource resource.Resource) error {
```

Comment:
Instead of having these methods return an error from `util.RunCmd` I may have them return an array of strings so I can unit test easier.